src/app/MessageDef/EventFilterIB.h

```diff
@@ -0,0 +1,146 @@
+namespace app {
+namespace EventFilterIB {
+enum class Tag : uint8_t
+{
+    kNode     = 0,
+    kEventMin = 1,
+};
+
+class Parser : public chip::app::Parser
+{
+public:
+    /**
+     *  @brief Initialize the parser object with TLVReader
+     *
+     *  @param [in] aReader A pointer to a TLVReader, which should point to the beginning of this EventFilterIB
+     *
+     *  @return #CHIP_NO_ERROR on success
+     */
+    CHIP_ERROR Init(const chip::TLV::TLVReader & aReader);
+
+#if CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
+    /**
+     *  @brief Roughly verify the message is correctly formed
+     *   1) all mandatory tags are present
+     *   2) all elements have expected data type
+     *   3) any tag can only appear once
+     *   4) At the top level of the structure, unknown tags are ignored for forward compatibility
+     *  @note The main use of this function is to print out what we're
+     *    receiving during protocol development and debugging.
+     *    The encoding rule has changed in IM encoding spec so this
+     *    check is only "roughly" conformant now.
+     *
+     *  @return #CHIP_NO_ERROR on success
+     */
+    CHIP_ERROR CheckSchemaValidity() const;
+#endif
+
+    /**
+     *  @brief Get a TLVReader for the NodeId. Next() must be called before accessing them.
+     *
+     *  @param [in] apNodeId    A pointer to apNodeId
+     *
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_ERROR_WRONG_TLV_TYPE if there is such element but it's not any of the defined unsigned integer types
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetNode(NodeId * const apNodeId) const;
+
+    /**
+     *  @brief Get a TLVReader for the ClusterId. Next() must be called before accessing them.
+     *
+     *  @param [in] apEventMin    A pointer to EventMin
+     *
+     *  @return #CHIP_NO_ERROR on success
+     *          #CHIP_ERROR_WRONG_TLV_TYPE if there is such element but it's not any of the defined unsigned integer types
+     *          #CHIP_END_OF_TLV if there is no such element
+     */
+    CHIP_ERROR GetEventMin(uint64_t * const apEventMin) const;
+};
+
+class Builder : public chip::app::Builder
+{
+public:
+    /**
+     *  @brief Initialize a EventFilterIB::Builder for writing into a TLV stream
+     *
+     *  @param [in] apWriter    A pointer to TLVWriter
+     *
+     *  @return #CHIP_NO_ERROR on success
+     */
+    CHIP_ERROR Init(chip::TLV::TLVWriter * const apWriter);
+
+    /**
+     * Init the EventFilterIB container with an particular context tag.
+     * Required to implement arrays of arrays, and to test ListBuilder.
+     *
+     * @param[in]   apWriter    Pointer to the TLVWriter that is encoding the message.
+     * @param[in]   aContextTagToUse    A contextTag to use.
+     *
+     * @return                  CHIP_ERROR codes returned by chip::TLV objects.
+     */
+    CHIP_ERROR Init(TLV::TLVWriter * const apWriter, const uint8_t aContextTagToUse);
```

Comment:
This is not implemented, but is the functionality we actually need, no?

Reply:
removed. thanks